file-sharing-back/file-sharing-back-core/src/main/java/com/file/sharing/core/dao/FileItemDao.java

```diff
@@ -12,4 +15,10 @@
 
 	PageResult<FileItem> getItemsBasedOnCriteria(ItemSearch itemSearch);
 
+	Optional<Long> sumOfAllUserFiles(Integer userId);
+
+    List<String> getSimilarFileItemsFromSameDirectory(FileItem fileItem);
```

Comment:
I don;t know if it's possible, but we can use getItemsBasedOnCriteria to perform any custom query search. Check if it's possible, and maybe if we can enhance the ItemSearch object and the getItemsBasedOnCriteria implementation to support regex search also

Reply:
we're doing it my way!!